### DISCUSSION TITLE
Can't download only the video, without audio

### DISCUSSION THREAD

**Discussion by XangelMusic** [aclap-dev/video-downloadhelper#1135] [Help and Q&A]
### Your question

How do you download only the video without the audio?

I plan to aggregate video and audio tracks for this one video because the main download failed. I am able to download just the audio track, but not a video track.

### Addon & CoApp version and details

version: 9.1.0.42
channel: stable
build date: 2024-08-02
build options: linuxlic: false, noyt: false, browser: firefox
lang: en-US
license: accepted
key (only 16 first characters): 75052b43f0614124
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:129.0) Gecko/20100101 Firefox/129.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\Josue Cerna",
  "converterBinary": "C:\\Program Files\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360955"
}

**Comment by Mattmjs** [comment]
Where is the problem happening ? , do you have a link to the video.

**Reply by XangelMusic** [reply]
https://virtual.spwn.jp/events/24080313-lvkowairoentropy/streaming

that's the link, but its a paid ticket to view it. basically VDH doesn't have permission to download from SPWN they say, but I remember downloading only just the video.

EDIT: I actually did send a report today using the new interface about the download error

**Reply by Mattmjs** [reply]
Ok I can't view it like you said. Next to download is the 3 dot button. Click this and look down to details.
Click on it and it'll open a new page with more information about the video. Copy this as text ( not an image ) and paste it here.

**Reply by XangelMusic** [reply]
Tab | 19 (incognito: false)
--|--
Title | SPWN
Page URL | https://virtual.spwn.jp/events/24080313-lvkowairoentropy/streaming with 11 headers
Thumbnail | https://public.spwn.jp/event-pages/banner_ogp.jpg
Low Quality | false
#0 main url | https://vod.spwn.jp/spwn-vod/24080313-lvkowairoentropy/grp2/cam1_v1/cam1_v1_.m3u8
#0 video component | https://vod.spwn.jp/spwn-vod/24080313-lvkowairoentropy/grp2/cam1_v1/cam1_v1_video_1080P_.m3u8
#0 audio component | https://vod.spwn.jp/spwn-vod/24080313-lvkowairoentropy/grp2/cam1_v1/cam1_v1_audio_320kbps_48k_.m3u8
#0 core media | Mp4 b:Hls p:hls 6083s
#0 audio | AAC at unknown
#0 video | H264 at 9322545, fps: 29.97, q:1080, {"width":1920,"height":1080}
#1 main url | https://vod.spwn.jp/spwn-vod/24080313-lvkowairoentropy/grp2/cam1_v1/cam1_v1_.m3u8
#1 video component | https://vod.spwn.jp/spwn-vod/24080313-lvkowairoentropy/grp2/cam1_v1/cam1_v1_video_720P_.m3u8
#1 audio component | https://vod.spwn.jp/spwn-vod/24080313-lvkowairoentropy/grp2/cam1_v1/cam1_v1_audio_320kbps_48k_.m3u8
#1 core media | Mp4 b:Hls p:hls 6083s
#1 audio | AAC at unknown
#1 video | H264 at 4429496, fps: 29.97, q:720, {"width":1280,"height":720}
#2 main url | https://vod.spwn.jp/spwn-vod/24080313-lvkowairoentropy/grp2/cam1_v1/cam1_v1_.m3u8
#2 video component | https://vod.spwn.jp/spwn-vod/24080313-lvkowairoentropy/grp2/cam1_v1/cam1_v1_video_480P_.m3u8
#2 audio component | https://vod.spwn.jp/spwn-vod/24080313-lvkowairoentropy/grp2/cam1_v1/cam1_v1_audio_320kbps_48k_.m3u8
#2 core media | Mp4 b:Hls p:hls 6083s
#2 audio | AAC at unknown
#2 video | H264 at 2207026, fps: 29.97, q:480, {"width":854,"height":480}

**Reply by GrampaWildWilly** [reply]
Rats.  All the m3u8 files in those details are not the expected HLS master or stream manifests.  They are actually XML files that contain an error message.  I suspect this means the web site has implemented security features to disallow hackers like us to get into the information that would allow us to download the content without actually being logged in members of the site.  It looks like you're not going to get any help here.

However, the item marked `#0 video component` is the video track.  If you are logged into the site, you should be able to click that link yourself & you will get access to the content.  When you click an HLS stream manifest like that, Firefox will give you a prompt asking you what to do with it, open it with Firefox or some other application, save it to your system, etc.  Simply dismiss the dialog.  You will be presented with an empty browser tab.  But with any luck, VDH will have detected something.  See if you can download the video track with VDH.

Similarly, `#0 audio component` is the audio track.  If you want that, you can get it the same way as the video track.

**Reply by Mattmjs** [reply]
I can't do anything with those m3u8 links, not in the browser or directly in ffmpeg. Looks like it will only work if you've paid
to view it. In that case you've sent the report which the dev will have received it.

If you wanted to get the video on it's own, look at each of the video component urls. There is a 1080p, 720p and 480p.
Try opening one of those in a new tab and see if VDH detects it since your paying to view it. All I get is

`This XML file does not appear to have any style information associated with it. The document tree is shown below.
<Error>
<Code>MissingKey</Code>
<Message>
Missing Key-Pair-Id query parameter or cookie value
</Message>
</Error>`

**Reply by XangelMusic** [reply]
It works! VDH can detect the video and audio components when opened in another tab and shown as a RawHls. Now I want to know how do I aggregate the video and audio using the updated UI?

**Reply by Mattmjs** [reply]
It is in the legacy interface if you want to do that, it's been suggested that it's a feature that could disappear at some point.
Because it is hardly used and not necessary anymore, since the software now downloads video and audio together as a single file.

**Reply by GrampaWildWilly** [reply]
ffmpeg -i videotrackfile -i audiotrackfile -codec: copy mergedvideoandaudiofile

But I would expect that if, instead of one of the component items in the details, you use one of the main items, you will get the video & audio already merged.

**Reply by XangelMusic** [reply]
I should note that when I tried to do the aggregate from the legacy UI it gave me this error:

```
[in#0 @ 0000025748cfb740] Error opening input: Invalid data found when processing input

Error opening input file E:\Users\Josue Cerna\Videos\dwhelper\vod spwn jpspwn-vod24080313-lvkowairoentropygrp2cam1v1cam1v1vide.mp4.

Error opening input files: Invalid data found when processing input
```

Anyone know what this error is about?

Although, I did merge the video and audio with the command as suggested by GrampaWildWilly and it works. But anyways, I got what I wanted and learned something new today. Thanks for the help!

**Reply by GrampaWildWilly** [reply]
Do I understand correctly that you are trying to use the `aggregate video with audio` function in the legacy UI?  The file that has the error opening would likely be one of the tracks of the original, either the video or the audio.  Can't tell which from what you've given.  But evidently, it isn't what ffmpeg is expecting.  I say ffmpeg because that's what's hiding inside the aggregate function.  It might just be an empty file.  I think it's not worth pursuing since you've got things working via other means.
